### PR TITLE
Note different .pubxml handling by VS and CLI

### LINKED ISSUE #34017: dotnet-publish: Which "publish-related properties" are supported?
[Here](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-publish#msbuild) the docs say:

```
You can also set publish-related properties by referring to a .pubxml file. For example:
dotnet publish -p:PublishProfile=FolderProfile
```

I would have assumed for instance that the `DeleteExistingFiles` property inside .pubxml is a _publish-related property_ but it is ignored by `dotnet publish`.

Also the `LastUsedBuildConfiguration` seems to be ignored because `dotnet publish` still builds and copies the Debug dlls, not the Release dlls although I have specified:

```
<LastUsedBuildConfiguration>Release</LastUsedBuildConfiguration>
```

Is `LastUsedBuildConfiguration` not a _publish-related property_? I would have guessed that it is.

I think it's not a mistake on my end because the `PublishDir` setting in .pubxml is honored when I do `dotnet publish` and the `DeleteExistingFiles` and `LastUsedBuildConfiguration` settings are honored when publishing via Visual Studio. Maybe `DeleteExistingFiles` and `LastUsedBuildConfiguration` are meant to only be used by Visual Studio but the docs here only mention `PublishUrl` to be meant to be used by Visual Studio.

I think it would be very helpful to write in the docs which _publish-related properties_ exactly are supported by `dotnet publish` in order to avoid confusion and maybe also it should be mentioned what the necessary replacements are for unsupported _publish-related properties_, e.g. `--configuration` argument has to be used instead of `LastUsedBuildConfiguration` setting.

Using .NET 7 SDK and tested with a completely new ASP.NET Core Web Application project. I didn't test any other _publish-related properties_.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 211afdc7-c796-202d-239d-823f2a499dff
* Version Independent ID: 9439897e-dcb3-bc2e-5cae-35c1ca2e0681
* Content: [dotnet publish command - .NET CLI](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-publish)
* Content Source: [docs/core/tools/dotnet-publish.md](https://github.com/dotnet/docs/blob/main/docs/core/tools/dotnet-publish.md)
* Product: **dotnet-fundamentals**
* Technology: **dotnet-cli**
* GitHub Login: @tdykstra
* Microsoft Alias: **tdykstra**

---
[Associated WorkItem - 66017](https://dev.azure.com/msft-skilling/Content/_workitems/edit/66017)

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #34017 

[Internal review URL](https://review.learn.microsoft.com/en-us/dotnet/core/tools/dotnet-publish?branch=pr-en-us-34269)

### CODE REVIEW

File: docs/core/tools/dotnet-publish.md [hunk 1]
```diff
@@ -65,6 +67,10 @@ In the *.pubxml* file:
 
 If you want the scenario to work in all places, you can initialize both these properties to the same value in the *.pubxml* file. When GitHub issue [dotnet/sdk#20931](https://github.com/dotnet/sdk/issues/20931) is resolved, only one of these properties will need to be set.
 
+Some properties in the *.pubxml* file are honored only by Visual Studio and have no effect on `dotnet publish`. We're working to bring the CLI more into alignment with Visual Studio's behavior. But some properties, such as `LastUsedBuildConfiguration`, may never be used by the CLI. The CLI and Visual Studio both do the packaging aspect of publishing, and [dotnet/sdk#29817](https://github.com/dotnet/sdk/pull/29817) plans to add support for more properties related to that. But the CLI doesn't do the deployment automation aspect of publishing, and properties related to that are not supported.
```
Comment (baronfel) [inline docs/core/tools/dotnet-publish.md:70]: For docs purposes, I'd want to reach out to @vijayrkn and team and see if there were examples of properties that are 100% known to be irrelevant to the CLI.

Comment (vijayrkn) [reply docs/core/tools/dotnet-publish.md:70]: The most notable ones are properties that represent configuration/platform/Runtime/Target identifiers that impact the build.

`LastUsedBuildConfiguration`
`Configuration`
`Platform`
`TargetFramework`
`TargetFrameworks`
`RuntimeIdentifier`

Comment (tdykstra) [reply docs/core/tools/dotnet-publish.md:70]:
```suggestion
Some properties in the *.pubxml* file are honored only by Visual Studio and have no effect on `dotnet publish`. We're working to bring the CLI more into alignment with Visual Studio's behavior. But some properties will never be used by the CLI. The CLI and Visual Studio both do the packaging aspect of publishing, and [dotnet/sdk#29817](https://github.com/dotnet/sdk/pull/29817) plans to add support for more properties related to that. But the CLI doesn't do the deployment automation aspect of publishing, and properties related to that are not supported. The most notable *.pubxml* properties that are not supported by `dotnet publish` are the ones that impact the build, such as the following properties:

* `LastUsedBuildConfiguration`
* `Configuration`
* `Platform`
* `TargetFramework`
* `TargetFrameworks`
* `RuntimeIdentifier`
```

File: docs/core/tools/dotnet-publish.md [hunk 2]
```diff
@@ -65,6 +67,17 @@ In the *.pubxml* file:
 
 If you want the scenario to work in all places, you can initialize both these properties to the same value in the *.pubxml* file. When GitHub issue [dotnet/sdk#20931](https://github.com/dotnet/sdk/issues/20931) is resolved, only one of these properties will need to be set.
 
+Some properties in the *.pubxml* file are honored only by Visual Studio and have no effect on `dotnet publish`. We're working to bring the CLI more into alignment with Visual Studio's behavior. But some properties will never be used by the CLI. The CLI and Visual Studio both do the packaging aspect of publishing, and [dotnet/sdk#29817](https://github.com/dotnet/sdk/pull/29817) plans to add support for more properties related to that. But the CLI doesn't do the deployment automation aspect of publishing, and properties related to that are not supported. The most notable *.pubxml* properties that are not supported by `dotnet publish` are the ones that impact the build, such as the following properties:
```
Comment (tdykstra) [inline docs/core/tools/dotnet-publish.md:70]:
```suggestion
Some properties in the *.pubxml* file are honored only by Visual Studio and have no effect on `dotnet publish`. We're working to bring the CLI more into alignment with Visual Studio's behavior. But some properties will never be used by the CLI. The CLI and Visual Studio both do the packaging aspect of publishing, and [dotnet/sdk#29817](https://github.com/dotnet/sdk/pull/29817) plans to add support for more properties related to that. But the CLI doesn't do the deployment automation aspect of publishing, and properties related to that are not supported. The most notable *.pubxml* properties that are not supported by `dotnet publish` are the following ones that impact the build:
```

File: docs/core/tools/dotnet-publish.md [hunk 3]
```diff
@@ -65,6 +67,17 @@ In the *.pubxml* file:
...
 If you want the scenario to work in all places, you can initialize both these properties to the same value in the *.pubxml* file. When GitHub issue [dotnet/sdk#20931](https://github.com/dotnet/sdk/issues/20931) is resolved, only one of these properties will need to be set.
 
+Some properties in the *.pubxml* file are honored only by Visual Studio and have no effect on `dotnet publish`. We're working to bring the CLI more into alignment with Visual Studio's behavior. But some properties will never be used by the CLI. The CLI and Visual Studio both do the packaging aspect of publishing, and [dotnet/sdk#29817](https://github.com/dotnet/sdk/pull/29817) plans to add support for more properties related to that. But the CLI doesn't do the deployment automation aspect of publishing, and properties related to that are not supported. The most notable *.pubxml* properties that are not supported by `dotnet publish` are the following ones that impact the build:
+
+* `LastUsedBuildConfiguration`
+* `Configuration`
+* `Platform`
+* `TargetFramework`
+* `TargetFrameworks`
+* `RuntimeIdentifier`
```
Comment (vijayrkn) [inline docs/core/tools/dotnet-publish.md:77]: For completeness, we should also add LastUsedPlatform& RuntimeIdentifiers (https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#runtimeidentifiers) along with RuntimeIdentifier(similar to TargetFrameworks & TargetFramework). Sorry, missing it in my initial email.

Comment (tdykstra) [inline docs/core/tools/dotnet-publish.md:77]:
```suggestion
* `RuntimeIdentifier`
*  `RuntimeIdentifiers` 
```

File: docs/core/tools/dotnet-publish.md [hunk 4]
```diff
@@ -65,6 +67,17 @@ In the *.pubxml* file:
 
 If you want the scenario to work in all places, you can initialize both these properties to the same value in the *.pubxml* file. When GitHub issue [dotnet/sdk#20931](https://github.com/dotnet/sdk/issues/20931) is resolved, only one of these properties will need to be set.
 
+Some properties in the *.pubxml* file are honored only by Visual Studio and have no effect on `dotnet publish`. We're working to bring the CLI more into alignment with Visual Studio's behavior. But some properties will never be used by the CLI. The CLI and Visual Studio both do the packaging aspect of publishing, and [dotnet/sdk#29817](https://github.com/dotnet/sdk/pull/29817) plans to add support for more properties related to that. But the CLI doesn't do the deployment automation aspect of publishing, and properties related to that are not supported. The most notable *.pubxml* properties that are not supported by `dotnet publish` are the following ones that impact the build:
+
+* `LastUsedBuildConfiguration`
+* `Configuration`
+* `Platform`
```
Comment (tdykstra) [inline docs/core/tools/dotnet-publish.md:74]:
```suggestion
* `Platform`
* `LastUsedPlatform`
```

**Review by CamSoper — APPROVED** [review]
LGTM

File: docs/core/tools/dotnet-publish.md [hunk 5]
```diff
@@ -65,6 +67,19 @@ In the *.pubxml* file:
...
+* `LastUsedPlatform`
+* `TargetFramework`
+* `TargetFrameworks`
+* `RuntimeIdentifier`
+*  `RuntimeIdentifiers`
```
Comment (tdykstra) [inline docs/core/tools/dotnet-publish.md:79]:
```suggestion
* `RuntimeIdentifiers`
```

**Pull request merged**